docs/guides/centos-installation.sh

```diff
@@ -0,0 +1,96 @@
+################################
+############ HackMD ############
+################################
+
+# Create your HackMD passphrase
+echo "Create your HackMD passphrase for the MySQL database and press [Enter]. You will create your specific HackMD credentials after the installation."
+read -s hackmdpassphrase
+
+# Set your IP address as a variable. This is for instructions below.
+IP="$(hostname -I | sed -e 's/[[:space:]]*$//')"
+
+# Install dependencies
+sudo yum install epel-release -y
+sudo yum install mariadb-server npm gcc-c++ git bzip2 -y
+
+# Stage HackMD for building
+sudo npm install -g uws node-gyp tap webpack grunt yarn
+sudo yarn add -D webpack-cli
+sudo git clone https://github.com/hackmdio/hackmd.git /opt/hackmd/
+cd /opt/hackmd
+sudo bin/setup
+cd -
+
+# Set up the HackMD database
+sudo systemctl start mariadb.service
+mysql -u root -e "CREATE DATABASE hackmd;"
+mysql -u root -e "GRANT ALL PRIVILEGES ON hackmd.* TO 'hackmd'@'localhost' IDENTIFIED BY '$hackmdpassphrase';"
+
+# Update the HackMD configuration files
+sudo sed -i 's/"username":\ ""/"username":\ "hackmd"/' /opt/hackmd/config.json
+sudo sed -i 's/"password":\ ""/"password":\ "'$hackmdpassphrase'"/' /opt/hackmd/config.json
+sudo sed -i 's/5432/3306/' /opt/hackmd/config.json
+sudo sed -i 's/postgres/mysql/' /opt/hackmd/config.json
+sudo sed -i 's/change\ this/mysql:\/\/hackmd:'$hackmdpassphrase'@localhost:3306\/hackmd/' /opt/hackmd/.sequelizerc
+
+# Build HackMD
+sudo npm run build --prefix /opt/hackmd/
```

Comment:
When done above this works also without sudo and prevents malicious submodules from hurting your system